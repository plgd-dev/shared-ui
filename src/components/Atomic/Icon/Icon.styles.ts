import { css } from '@emotion/react'

export const icon = css`
    display: flex;
    align-items: center;
    justify-content: center;
`
