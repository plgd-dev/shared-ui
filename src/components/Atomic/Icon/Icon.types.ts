import { FC, SVGProps } from 'react'

export type Props = {
    className?: string
    icon: string
    id?: string
    onCompleted?: (name: string, SvgIcon: FC<SVGProps<SVGSVGElement>> | undefined) => void
    onError?: (err: Error) => void
    size?: number
} & SVGProps<SVGSVGElement>

export const defaultProps = {
    size: 16,
}
