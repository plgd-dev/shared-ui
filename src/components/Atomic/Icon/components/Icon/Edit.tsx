// Auto-generated file created by svgr-cli source svg-template.js
// Run yarn icons:create to update
// Do not edit
import * as React from 'react'
import { SVGProps } from 'react'
const SvgEdit = (props: SVGProps<SVGSVGElement>) => (
    <svg width={props.width || 16} height={props.height || 16} viewBox='0 0 20 20' fill='none' xmlns='http://www.w3.org/2000/svg' {...props}>
        <path
            d='M17.178 2.207a1.84 1.84 0 0 0-2.605 0l-.99.987 3.22 3.22.99-.99a1.84 1.84 0 0 0 0-2.605l-.615-.612Zm-9.84 7.236c-.202.2-.356.447-.445.72l-.974 2.92a.793.793 0 0 0 .191.81c.21.214.523.286.81.19l2.92-.973c.27-.092.517-.243.72-.444l5.503-5.506-3.223-3.223-5.503 5.506ZM4.823 3.598a3.158 3.158 0 0 0-3.157 3.158v8.42a3.158 3.158 0 0 0 3.157 3.157h8.42a3.158 3.158 0 0 0 3.158-3.157v-3.158a1.051 1.051 0 1 0-2.105 0v3.158c0 .582-.47 1.052-1.053 1.052h-8.42c-.582 0-1.052-.47-1.052-1.052v-8.42c0-.582.47-1.053 1.052-1.053h3.158a1.051 1.051 0 1 0 0-2.105H4.824Z'
            fill='#81868C'
        />
    </svg>
)
export default SvgEdit
