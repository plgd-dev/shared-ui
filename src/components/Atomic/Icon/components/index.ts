export { default as IconActions } from './IconActions'
export { default as IconArrowDetail } from './IconArrowDetail'
export { default as IconArrowDownNoPadding } from './IconArrowDownNoPadding'
export { default as IconArrowDown } from './IconArrowDown'
export { default as IconArrowLeft } from './IconArrowLeft'
export { default as IconArrowRight } from './IconArrowRight'
export { default as IconArrowTriangleFullUp } from './IconArrowTriangleFullUp'
export { default as IconBell } from './IconBell'
export { default as IconCertificate } from './IconCertificate'
export { default as IconChat } from './IconChat'
export { default as IconCloseCircle } from './IconCloseCircle'
export { default as IconClose } from './IconClose'
export { default as IconCloudError } from './IconCloudError'
export { default as IconCloudSuccess } from './IconCloudSuccess'
export { default as IconCloudWarning } from './IconCloudWarning'
export { default as IconCollapse } from './IconCollapse'
export { default as IconCopy } from './IconCopy'
export { default as IconDashboard } from './IconDashboard'
export { default as IconDeviceUpdate } from './IconDeviceUpdate'
export { default as IconDevices } from './IconDevices'
export { default as IconDocs } from './IconDocs'
export { default as IconDownload } from './IconDownload'
export { default as IconEdit } from './IconEdit'
export { default as IconError } from './IconError'
export { default as IconFilePem } from './IconFilePem'
export { default as IconFileUpload } from './IconFileUpload'
export { default as IconGlobe } from './IconGlobe'
export { default as IconHidePassword } from './IconHidePassword'
export { default as IconInfo } from './IconInfo'
export { default as IconIntegrations } from './IconIntegrations'
export { default as IconLink } from './IconLink'
export { default as IconLoader } from './IconLoader'
export { default as IconLock } from './IconLock'
export { default as IconLog } from './IconLog'
export { default as IconLogout } from './IconLogout'
export { default as IconNet } from './IconNet'
export { default as IconNetwork } from './IconNetwork'
export { default as IconNotification } from './IconNotification'
export { default as IconPendingCommands } from './IconPendingCommands'
export { default as IconPlus } from './IconPlus'
export { default as IconRefresh } from './IconRefresh'
export { default as IconRemoteClients } from './IconRemoteClients'
export { default as IconSearch } from './IconSearch'
export { default as IconSettings } from './IconSettings'
export { default as IconShield } from './IconShield'
export { default as IconShowPassword } from './IconShowPassword'
export { default as IconSortDown } from './IconSortDown'
export { default as IconSortUp } from './IconSortUp'
export { default as IconSuccess } from './IconSuccess'
export { default as IconTableArrowDown } from './IconTableArrowDown'
export { default as IconTableArrowUp } from './IconTableArrowUp'
export { default as IconToastClose } from './IconToastClose'
export { default as IconTrash } from './IconTrash'
export { default as IconWarning } from './IconWarning'
