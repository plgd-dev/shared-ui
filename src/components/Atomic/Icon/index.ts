export { default } from './Icon'
export * from './Icon'

export * from './components'

export * from './utils'
