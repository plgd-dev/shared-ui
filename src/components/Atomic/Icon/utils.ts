export const convertSize = (size = 16) => ({
    height: size,
    width: size,
})
