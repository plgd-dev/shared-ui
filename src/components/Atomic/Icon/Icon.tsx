import { FC, forwardRef, SVGProps, useEffect, useRef, useState } from 'react'
import { Props, defaultProps } from './Icon.types'
import * as styles from './Icon.styles'

export const Icon = forwardRef<any, Props>((props, ref) => {
    const { className, icon, id, size, onError, onCompleted, ...rest } = { ...defaultProps, ...props }

    const [loading, setLoading] = useState(false)
    const [iconLoad, setIconLoad] = useState('')
    const [error, setError] = useState<Error | undefined>(undefined)
    const ImportedIconRef = useRef<FC<SVGProps<SVGSVGElement>> | undefined>(undefined)

    const IconComponent = ImportedIconRef.current ? ImportedIconRef.current : () => null

    const parseComponentName = (string: string) =>
        string
            .split('-')
            .map((part) => part.charAt(0).toUpperCase() + part.slice(1))
            .join('')

    useEffect(() => {
        setLoading(true)
        const componentName = parseComponentName(icon)
        const importIcon = async (): Promise<void> => {
            try {
                await import(`./components/${componentName}.tsx`).then((IconComponent) => {
                    ImportedIconRef.current = IconComponent.default
                    return IconComponent.default
                })
                onCompleted?.(icon, ImportedIconRef.current)
            } catch (err: any) {
                onError?.(err)
                setError(err)
            } finally {
                setLoading(false)
            }
        }
        importIcon().then(() => {
            setIconLoad(icon)
        })
    }, [icon, onCompleted, onError])

    if (error) {
        console.error(error)
        return null
    }

    return (
        <div
            className={className}
            css={styles.icon}
            data-icon={iconLoad}
            id={id}
            ref={ref}
            style={{
                width: size,
                height: size,
            }}
        >
            {loading ? null : <IconComponent {...rest} height={size} width={size} />}
        </div>
    )
})

Icon.displayName = 'Icon'

export default Icon
