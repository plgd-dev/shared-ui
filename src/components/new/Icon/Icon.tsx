import { FC, SVGProps, useEffect, useRef, useState } from 'react'
import { Props, defaultProps } from './Icon.types'

export const Icon: FC<Props> = (props) => {
    const { className, icon, id, size, onError, onCompleted, ...rest } = { ...defaultProps, ...props }

    const [loading, setLoading] = useState(false)
    const [error, setError] = useState<Error | undefined>(undefined)
    const ImportedIconRef = useRef<FC<SVGProps<SVGSVGElement>> | undefined>(undefined)

    const IconComponent = ImportedIconRef.current ? ImportedIconRef.current : () => null

    const parseComponentName = (string: string) =>
        string
            .split('-')
            .map((part) => part.charAt(0).toUpperCase() + part.slice(1))
            .join('')

    useEffect(() => {
        setLoading(true)
        const componentName = parseComponentName(icon)
        const importIcon = async (): Promise<void> => {
            try {
                await import(`./components/${componentName}.tsx`).then((IconComponent) => {
                    ImportedIconRef.current = IconComponent.default
                    return IconComponent.default
                })
                onCompleted?.(icon, ImportedIconRef.current)
            } catch (err: any) {
                onError?.(err)
                setError(err)
            } finally {
                setLoading(false)
            }
        }
        importIcon().then()
    }, [icon, onCompleted, onError])

    if (error) {
        console.error(error)
        return null
    }

    return (
        <div
            className={className}
            id={id}
            style={{
                width: size,
                height: size,
            }}
        >
            {loading ? null : <IconComponent {...rest} height={size} width={size} />}
        </div>
    )
}

Icon.displayName = 'Icon'
Icon.defaultProps = defaultProps

export default Icon
