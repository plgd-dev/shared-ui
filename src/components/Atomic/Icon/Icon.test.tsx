import { waitFor } from '@testing-library/react'
import { render } from '../../../../test/jest-wrapper'
import Icon, { IconsRaw } from '../Icon'

describe('<Icon>', () => {
    it('render correctly - snapshot', async () => {
        const { asFragment } = render(
            <div>
                {IconsRaw.map((icon, index) => (
                    <Icon icon={icon} key={index} />
                ))}
            </div>
        )

        await waitFor(() => {
            expect(asFragment()).toMatchSnapshot()
        })
    })
})
