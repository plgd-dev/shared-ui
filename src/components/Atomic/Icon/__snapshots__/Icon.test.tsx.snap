// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<div>
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
    <div
      class="emotion-0"
      data-icon=""
      style="width: 16px; height: 16px;"
    />
  </div>
</DocumentFragment>
`;
